src/ddmd/backend/cgen.c

```diff
@@ -713,181 +713,76 @@ void regwithvalue(CodeBuilder& cdb,regm_t regm,targ_size_t value,unsigned *preg,
+ * within this module, we stuff it in an array of references to be
+ * fixed up later.
  */
-
-struct fixlist
+struct Fixup
 {
-    int         Lseg;           // where the fixup is going (CODE or DATA, never UDATA)
-    int         Lflags;         // CFxxxx
-    targ_size_t Loffset;        // addr of reference to symbol
-    targ_size_t Lval;           // value to add into location
+    symbol      *sym;       // the referenced symbol
+    int         seg;        // where the fixup is going (CODE or DATA, never UDATA)
+    int         flags;      // CFxxxx
+    targ_size_t offset;     // addr of reference to symbol
+    targ_size_t val;        // value to add into location
 #if TARGET_OSX
-    symbol      *Lfuncsym;      // function the symbol goes in
+    symbol      *funcsym;   // function the symbol goes in
 #endif
-    fixlist *Lnext;             // next in threaded list
-
-    static int nodel;           // don't delete from within searchfixlist
 };
 
-int fixlist::nodel = 0;
-
-/* The AArray, being hashed on the pointer value of the symbol s, is in a different
- * order from run to run. This plays havoc with trying to compare the .obj file output.
- * When needing to do that, set FLARRAY to 1. This will replace the AArray with a
- * simple (and very slow) linear array. Handy for tracking down compiler issues, though.
- */
-#define FLARRAY 0
-struct Flarray
+struct FixupArray
 {
-#if FLARRAY
-    symbol *s;
-    fixlist *fl;
-
-    static Flarray *flarray;
-    static size_t flarray_dim;
-    static size_t flarray_max;
+    Fixup *ptr;
+    size_t dim, cap;
 
-    static fixlist **add(symbol *s)
-    {
-        //printf("add %s\n", s->Sident);
-        fixlist **pv;
-        for (size_t i = 0; 1; i++)
-        {
-            assert(i <= flarray_dim);
-            if (i == flarray_dim)
-            {
-                if (flarray_dim == flarray_max)
-                {
-                    flarray_max = flarray_max * 2 + 1000;
-                    flarray = (Flarray *)mem_realloc(flarray, flarray_max * sizeof(flarray[0]));
-                }
-                flarray_dim += 1;
-                flarray[i].s = s;
-                flarray[i].fl = NULL;
-                pv = &flarray[i].fl;
-                break;
-            }
-            if (flarray[i].s == s)
-            {
-                pv = &flarray[i].fl;
-                break;
-            }
-        }
-        return pv;
-    }
+    FixupArray()
+    : ptr(NULL)
+    , dim(0)
+    , cap(0)
+    {}
 
-    static fixlist **search(symbol *s)
+    void push(const Fixup &e)
     {
-        //printf("search %s\n", s->Sident);
-        fixlist **lp = NULL;
-        for (size_t i = 0; i < flarray_dim; i++)
+        if (dim == cap)
         {
-            if (flarray[i].s == s)
-            {
-                lp = &flarray[i].fl;
-                break;
-            }
+            cap = cap ? 2 * cap : 16;
+            ptr = (Fixup *)::mem_realloc(ptr, cap * sizeof(Fixup));
```

Comment:
I bet if you put a print statement here, there will be a bunch of reallocations and copying. I suggest putting in a print, and then adjusting the initial size to be big enough for most compiles. 16 is way, way too small.